spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -110,11 +125,16 @@ public long lastModified() {
 	}
 
 	@Override
-	public Resource createRelative(String relativePath) throws IOException {
-		return new GoogleStorageResourceObject(this.storage,
+	public GoogleStorageResourceObject createRelative(String relativePath) throws IOException {
+		GoogleStorageResourceObject resource = new GoogleStorageResourceObject(this.storage,
 				getURI().toString() + (relativePath.startsWith("/") ? "" : "/")
 						+ relativePath,
-				this.createBucketIfRequired);
+				this.createBucketIfNotExists);
+		if (!resource.exists()) {
```

Comment:
I'm not sure that we should be creating the file if it doesn't exist. We don't do anything like this in the `GoogleStorageResourceObject` constructor. So, why would this be any different? I would just return the `resource` handle and let the user decide if they want to attempt creating it.

Reply:
It's the one thing this method must do and we're not doing currently: creating a file.
We've seen use cases that are broken if we don't create the file here. e.g., the one reported in #411

Reply:
#411 would be addressed by simply making `create()` public.

I'm not saying it's wrong, I'm just saying that it's inconsistent with the `GoogleStorageResourceObject` constructor. Why not create the file in the constructor, if doesn't exist? How is this different from `createRelative` from the bucket?

Reply:
`createSignedUrl()` would still not allow to create signed URLs for files that don't exist yet, so making `create()` public isn't enough.

Constructors shouldn't contain such complex logic as creating a file in a file system. The constructor Javadoc points out that the object is a "representation" of a file on GCS.
When I call `createRelative()`, I do expect the file to actually be created in GCS, not just in memory.

It isn't very much different from the bucket's `createRelative()`, except that you only need to have the bucket to call the bucket's, and if you have the file, you can call from it too.